### DISCUSSION TITLE
Filter on.event by event type?

### DISCUSSION THREAD

**Discussion by hapatrick** [nolar/kopf#1037] [Q&A] [answered]
Is there a way to filter calls to @kopf.on.event() handlers by event type? E.g. so the only handler is only called with events of type 'DELETED'?

I tried adding field='type' and value='DELETED', but that did not work. And I didn't really expect it to since I believe those filters are applied to fields of object that caused the event, not the event itself.

I don't see any way to do this, but maybe I'm missing something.

**Comment by nolar** [comment, marked as answer]
Using the callback filter “when=“?

**Reply by hapatrick** [reply]
Not sure why I didn't think of that. Thank you.
